ledger/sandbox/src/main/scala/com/digitalasset/platform/index/LedgerBackedIndexService.scala

```diff
@@ -143,8 +143,12 @@ abstract class LedgerBackedIndexService(
         party -> filters.inclusive.fold(Set.empty[Identifier])(_.templateIds)
     }
 
-  override def currentLedgerEnd(): Future[LedgerOffset.Absolute] =
-    Future.successful(toAbsolute(ledger.ledgerEnd))
+  override def currentLedgerEnd(): Future[LedgerOffset.Absolute] = {
+    if (ledger.ledgerEnd == Offset.beforeBegin) {
+      Future.successful(toAbsolute(Offset.begin))
+    } else
+      Future.successful(toAbsolute(ledger.ledgerEnd))
+  }
```

Comment:
Factoring

```suggestion
    Future.successful(toAbsolute(
      if (ledger.ledgerEnd == Offset.beforeBegin) Offset.begin
      else ledger.ledgerEnd
    ))
```

Reply:
```suggestion
    val offset =
      if (ledger.ledgerEnd == Offset.beforeBegin) Offset.begin
      else ledger.ledgerEnd
    Future.successful(toAbsolute(offset))
  }
```

Reply:
Hopefully Samir's suggestion works for both of you